django/db/backends/base/schema.py

```diff
@@ -767,19 +726,70 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
         # Drop the default if we need to
         # (Django usually does not use in-database defaults)
         if needs_database_default:
+            sql, params = self._alter_column_default_sql(model, old_field, new_field, True)
             sql = self.sql_alter_column % {
                 "table": self.quote_name(model._meta.db_table),
-                "changes": self.sql_alter_column_no_default % {
-                    "column": self.quote_name(new_field.column),
-                    "type": new_type,
-                }
+                "changes": sql,
             }
-            self.execute(sql)
+            self.execute(sql, params)
         # Reset connection if required
         if self.connection.features.connection_persists_old_columns:
             self.connection.close()
 
-    def _alter_column_type_sql(self, table, old_field, new_field, new_type):
+    def _alter_column_null_sql(self, model, old_field, new_field):
+        """
+        Hook to specialize column null alteration for different backends.
```

Comment:
I'd say "Return a (sql, params) fragment to set a column to null or non-null as required by new_field, or None if no changes are required."